### PR TITLE
test/e2e/framework/node/:remove TODO and and make some functions private

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref:https://github.com/kubernetes/kubernetes/issues/86763

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: test/e2e/framework/node/resource.go [hunk 1]
```diff
@@ -67,51 +67,40 @@ func FirstAddress(nodelist *v1.NodeList, addrType v1.NodeAddressType) string {
 	return ""
 }
 
-// TODO: better to change to a easy read name
```
Comment (tanjunchen) [inline test/e2e/framework/node/resource.go]: i think this name is ok,so i want to remove this TODO

File: test/e2e/framework/node/resource.go [hunk 2]
```diff
@@ -343,16 +329,16 @@ func GetMasterAndWorkerNodes(c clientset.Interface) (sets.String, *v1.NodeList,
...
 			nodes.Items = append(nodes.Items, n)
 		}
 	}
 	return masters, nodes, nil
 }
 
-// IsNodeUntainted tests whether a fake pod can be scheduled on "node", given its current taints.
+// isNodeUntainted tests whether a fake pod can be scheduled on "node", given its current taints.
 // TODO: need to discuss wether to return bool and error type
-func IsNodeUntainted(node *v1.Node) bool {
```
Comment (tanjunchen) [inline test/e2e/framework/node/resource.go]: this function is used only test/e2e/framework/node , so make it private.

File: test/e2e/framework/node/resource.go [hunk 3]
```diff
@@ -138,8 +126,8 @@ func IsConditionSetAsExpectedSilent(node *v1.Node, conditionType v1.NodeConditio
 	return isNodeConditionSetAsExpected(node, conditionType, wantTrue, true)
 }
 
-// IsConditionUnset returns true if conditions of the given node do not have a match to the given conditionType, otherwise false.
-func IsConditionUnset(node *v1.Node, conditionType v1.NodeConditionType) bool {
```
Comment (tanjunchen) [inline test/e2e/framework/node/resource.go]: this function is used only test/e2e/framework/node , so make it private.

File: test/e2e/framework/node/resource.go [hunk 4]
```diff
@@ -67,51 +67,40 @@ func FirstAddress(nodelist *v1.NodeList, addrType v1.NodeAddressType) string {
 	return ""
 }
 
-// TODO: better to change to a easy read name
 func isNodeConditionSetAsExpected(node *v1.Node, conditionType v1.NodeConditionType, wantTrue, silent bool) bool {
 	// Check the node readiness condition (logging all).
+	// Ensure that the condition type and the status matches as desired.
 	for _, cond := range node.Status.Conditions {
-		// Ensure that the condition type and the status matches as desired.
-		if cond.Type == conditionType {
```
Comment (tanjunchen) [inline test/e2e/framework/node/resource.go]: For line 75 and 77, This looks a little weird.
function IsConditionSetAsExpected and IsConditionSetAsExpectedSilent use isNodeConditionSetAsExpected,
and all values passed to conditionType are "NodeReady"

Comment (neolit123) [reply test/e2e/framework/node/resource.go]: the two different function seem to wrap this one while passing `silent=true/false`.

but the diff here removes the check against conditionType, which makes the name of the function invalid.
i suggest this function is kept like that for now and only try to remove this TODO:

> // TODO: better to change to a easy read name

File: test/e2e/framework/node/resource.go [hunk 5]
```diff
@@ -67,51 +67,40 @@ func FirstAddress(nodelist *v1.NodeList, addrType v1.NodeAddressType) string {
...
+					msg = fmt.Sprintf("Condition %s of node %s is %v, but Node is tainted by NodeController with %v. Failure",
+						conditionType, node.Name, cond.Status == v1.ConditionTrue, taints)
+				}
+				if !silent {
+					e2elog.Logf(msg)
 				}
 				return false
 			}
-			if (wantTrue && (cond.Status == v1.ConditionTrue)) || (!wantTrue && (cond.Status != v1.ConditionTrue)) {
+			// TODO: check if the Node is tainted once we enable NC notReady/unreachable taints by default
```
Comment (neolit123) [inline test/e2e/framework/node/resource.go]: this TODO is for sig-node to resolve if they want to.
taint based eviction is enabled by default in 1.13:
https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/#taint-based-evictions

Comment (tanjunchen) [reply test/e2e/framework/node/resource.go]: yes,i want to resolve this if i can in later.

File: test/e2e/framework/node/resource.go (hunk 4 above)
Comment (tanjunchen) [reply test/e2e/framework/node/resource.go]: thanks your review. The code implementation here always feels a bit weird i feel.
how about this change now?

Comment (neolit123) [reply test/e2e/framework/node/resource.go]: why modify the contents of the function?

Comment (tanjunchen) [reply test/e2e/framework/node/resource.go]: see https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/node/resource.go#L75 , 
 if cond.Type == conditionType  is true and cond.Type == v1.NodeReady is true, https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/node/resource.go#L114 never run。
I think it is possible to judge this condition in advance, so move this code. If you don't think need to change it, I can revert it.I can just try to remove this TODO.

Comment (neolit123) [reply test/e2e/framework/node/resource.go]: so in this PR i think we should only:
- removed the TODO `// TODO: better to change to a easy read name`.
- make the functions `is....()` private.

leave this TODO:
> // TODO: check if the Node is tainted once we enable NC notReady/unreachable taints by default

to SIG Node and not touch it.

Comment (tanjunchen) [reply test/e2e/framework/node/resource.go]: ok

Comment (tanjunchen) [reply test/e2e/framework/node/resource.go]: done,updated

**Review by neolit123 — COMMENTED** [review]
thanks
/lgtm
/approve

**Pull request merged**